services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsService.java

```diff
@@ -190,8 +192,20 @@ protected final CacheDirective getCacheDirective(final String type, final long m
                 });
     }
 
+    private static CredentialsResult<JsonObject> overrideAuthIdWithGeneratedAuthIdIfExists(
+            final CredentialsResult<JsonObject> credentialsResult) {
+        if (!credentialsResult.isError()) {
+            final JsonObject credential = credentialsResult.getPayload();
+            Optional.ofNullable(credential)
+                    .map(c -> c.getString(RegistryManagementConstants.FIELD_TYPE))
+                    .filter(RegistryManagementConstants.SECRETS_TYPE_X509_CERT::equals)
+                    .map(ok -> credential.getString(RegistryManagementConstants.FIELD_GENERATED_AUTH_ID))
+                    .ifPresent(id -> credential.put(RegistryManagementConstants.FIELD_AUTH_ID, id));
```

Comment:
Since this is the Credentials service, I would assume that all these constants should be defined in CredentialsConstants, right?